### DISCUSSION TITLE
Template for components pages

### DISCUSSION THREAD

**Discussion by tomyems** [UKHomeOffice/design-system#430] [Ideas]
The components pages have different and inconsistent heading structures. The proposal is to align them to have the same structure:

Heading 1: Component name
One line introduction, followed by example with Props.

Heading 2: When to use this component
More detailed description about how this component solves a problem.

Heading 2: How it works
Detailed examples showing different ways of using the component, including HTML and React code snippets.

Heading 2: How to use this component
Describe the best practice for the text used, and how it interacts with the rest of the page.

Heading 2: Accessibility
Guidance about how to use the component accessibly.

Heading 2: Research
Any research insights, including services currently using the component.

Heading 2: Contribute
Link to github discussion about the component, pointing out any known gaps.

**Comment by daniel-ac-martin** [comment]
One thing to consider, would be whether we want to align to or deviate from GDS. (I think we should be _fairly_ close at the moment. And what you've got above looks pretty close too.)

**Reply by daniel-ac-martin** [reply]
FWIW, In the GDS version I _think_ "How to use this component" would be blended into "How it works".

**Reply by daniel-ac-martin** [reply]
Sometimes we might not actually need the accessibility heading.

**Comment by Sarah-Lombardi** [comment]
Discussed at design system working group. Wondered if 'Contribute' was not as inviting as 'Help improve this component'. And consider any standard wording if we haven't got content yet for the research bit.

**Comment by tomyems** [comment]
Suggested improvment

Heading 1: Component name
One line introduction, followed by example with Props.

Heading 2: When to use this component
More detailed description about how this component solves a problem.

Heading 2: How it works
Detailed examples showing different ways of using the component, including HTML and React code snippets.

Heading 2: How to use this component
Describe the best practice for the text used, and how it interacts with the rest of the page. Suggested sub-headings might be 'Writing for this component', 'Placement on the page'.

Heading 2: Accessibility
Guidance about how to use the component accessibly.

Heading 2: Research
Any research insights, including services currently using the component.

Heading 2: Help improve this component
Link to github discussion about the component, pointing out any known gaps.

**Reply by tomyems** [reply]
Feedback about needing a 'Writing for this component' as a heading level 2, and moving the placement section under 'How it works'.

**Comment by tomyems** [comment]
Heading 1: Component name
One line introduction, followed by example with Props.

Heading 2: When to use this component
More detailed description about how this component solves a problem.

Heading 2: How it works
Detailed examples showing different ways of using the component, including HTML and React code snippets.
If helpful, include subheadings for the placement of the component on the page, and how it interacts with other components.

Heading 2: Writing for this component
Describe the best practice for the text used.

Heading 2: Accessibility
Guidance about how to use the component accessibly.

Heading 2: Research
Any research insights, including services currently using the component.

Heading 2: Help improve this component
Link to github discussion about the component, pointing out any known gaps.

**Comment by tomyems** [comment]
The changes to standardise the component pages can be viewed here: https://62f3df93fa4c5d0d2b9cd63e--hods.netlify.app/components

(See the pull request https://github.com/UKHomeOffice/design-system/pull/444)

**Comment by Sarah-Lombardi** [comment]
There could be an overlap between accessibility and research, particularly with the page example where the research referenced was mostly on accessibility. Might there be times where you can merge headers or drop one?

**Comment by Sarraounia** [comment]
I'd agree with @Sarah-Lombardi that there should be some flex in what headers are included, to help with clarity for individual components. 

@tomyems will this guidance be stored somewhere internally (GitHub and/or Sharepoint), so that if we create new pages in future we can follow the guidelines?

**Reply by tomyems** [reply]
My plan so far was to keep this discussion open, to use it as the record for what headers should be included. If that doesn't work, or you think it's better in another place, I'm all ears.

**Comment by tomyems** [comment]
Latest version: https://6304ccddfb22d77dee029268--hods.netlify.app/components
